### ISSUE TITLE
Update PHP 8 uploadprogress "Missing arginfo for uploadprogress_get_info()"

### ISSUE THREAD

**Issue by Jesmaster** [drud/ddev#3278]
After updating to 1.18.0 I noticed my "ddev composer" calls were coming back with new warning messages:

`PHP Warning:  Missing arginfo for uploadprogress_get_info() in Unknown on line 0`
`PHP Warning:  Missing arginfo for uploadprogress_get_contents() in Unknown on line 0`

I googled this and found the PHP bug ticket and the fix which is to update the uploadprogress library to version 1.1.4.  
The current vesion for PHP 8 in the latest ddev is 1.1.3.

https://bugs.php.net/bug.php?id=81320

**Comment by rfay** [comment]
It's annoying, and will be updated as soon as it hits the deb.sury.org packages. Has finally been fixed upstream though.

**Comment by mrweiner** [comment]
This will be fixed when what hits deb.sury.org packages? I'm seeing the same issue.

**Comment by rfay** [comment]
The problem is https://bugs.php.net/bug.php?id=81320 - it has been fixed upstream, but it has not yet made its way into the deb.sury.org packages that ddev-webserver uses. You can work around it by using PHP 7.4, which I don't think has this problem.

**Comment by mrweiner** [comment]
I see. No dice on 7.4 if using 8.0 syntax unfortunately. Unfortunately this does break some things like `ddev drush sql-sync`, but not much that can be done if it's an upstream issue .

**Comment by rfay** [comment]
You can upgrade it with a .ddev/web-build/Dockerfile. Try this and let me know how it goes.

```dockerfile
ARG BASE_IMAGE
FROM $BASE_IMAGE

RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg::Options::="--force-confnew" --no-install-recommends --no-install-suggests build-essential make autoconf libc-dev pkg-config php-pear php${PHP_VERSION}-dev 
RUN apt-get remove -y php8.0-uploadprogress php-uploadprogress
# The "echo" below just forces accepting the "automatic" configuration, the same as hitting <RETURN>
RUN echo | sudo pecl install uploadprogress
```

You can enable it with `sudo phpenmod uploadprogress`

**Comment by rfay** [comment]
@mrweiner you know you can just disable uploadprogress if you're not using it, right? `sudo phpdismod uploadprogress`

**Comment by mrweiner** [comment]
@rfay I'm getting `E: Failed to fetch` for the sury package URLs using that Dockerfile, but I didn't even think to just disable it. That works fine for my purposes. Thanks for the help

**Comment by rfay** [comment]
If this isn't fixed upstream by the next point release I'll just disable uploadprogress on php8 for the time being.

**Comment by rfay** [comment]
Requested in https://github.com/oerdnj/deb.sury.org/issues/1673

**Comment by weitzman** [comment]
Thanks for herding cats.

**Comment by rfay** [comment]
Good news, deb.sury.org has upgraded uploadprogress.

**Comment by drupal-daffie** [comment]
I am still getting the errors. What do I need to do to get rid of them? Which cache do I need to clear?

**Comment by rfay** [comment]
You will get them until next release, but if you want to fix earlier you can do the suggestion above (https://github.com/drud/ddev/issues/3278#issuecomment-935181406) or you can just add a `sudo apt-get update && sudo apt-get upgrade -y` to your post-start hook or to a .ddev/web-build/Dockerfile.

**Comment by drupal-daffie** [comment]
@rfay: Thank you for your help.

I have tried both solutions and the one with the custom dockerfile is a lot quicker then the one with post-start hook. The only small problem that I have is that both solutions do not work in combination with a PostgreSQL database.
Any idea when there will be a new release that fixes the problem with uploadprogress?

**Comment by rfay** [comment]
@drupal-daffie

* Do you need uploadprogress? If not, just disable it, either in dockerfile or post-start hook. `phpdismod uploadprogress`
* I can't understand what uploadprogress would have to do with postgres. Your postgres container should not have php installed in it. 

I'm just waiting on release now for xdebug to be supported with php8.1. That probably means when php8.1 gets full release in a couple of weeks. It really is time.

**Comment by drupal-daffie** [comment]
The DDEV environment with PostgreSQL works fine with PHP 7.4.

I work on Drupal Core and some tests need to uploadprogess extension.

**Comment by rfay** [comment]
Right, but uploadprogress works fine on php7.4 doesn't it? I thought the regression was only for 8.0

**Comment by rfay** [comment]
Closing because this is fixed in DDEV v1.18.1, which I'm working on right now. Should be out in the next few hours. Thanks for your patience.

**Comment by rfay** [comment]
Sorry, I ran into a blocker for release, but this will get in there in the next few days.
